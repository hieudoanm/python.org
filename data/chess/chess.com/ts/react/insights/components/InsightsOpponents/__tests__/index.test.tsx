import { render } from '@testing-library/react';
import { Insights } from '@web/services/chess/chess.dto';
import { mockResizeObserver } from '@web/utils/mock-resize-observer';
import { InsightsOpponents } from '..';

describe('InsightsOpponents', () => {
  beforeEach(() => {
    mockResizeObserver();
  });

  it('to match snapshot', () => {
    const { container } = render(
      <InsightsOpponents insights={{ opponents: [] } as unknown as Insights} />
    );
    expect(container).toMatchSnapshot();
  });
});
