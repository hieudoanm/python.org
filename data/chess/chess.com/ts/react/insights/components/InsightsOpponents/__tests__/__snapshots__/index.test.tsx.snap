// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsOpponents to match snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-y-2 text-center"
    id="opponents"
  >
    <div
      class="text-lg md:text-xl lg:text-2xl"
    >
      <div
        class="flex items-center justify-center gap-x-2"
      >
        <svg
          class="text-teal-500"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M399 384.2C376.9 345.8 335.4 320 288 320l-64 0c-47.4 0-88.9 25.8-111 64.2c35.2 39.2 86.2 63.8 143 63.8s107.8-24.7 143-63.8zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm256 16a72 72 0 1 0 0-144 72 72 0 1 0 0 144z"
          />
        </svg>
         Opponents
      </div>
    </div>
    <p
      class="text-xs md:text-sm lg:text-base"
    >
      Most frequently played opponents
    </p>
  </div>
  <div
    class="card border-base-content border shadow"
  >
    <div
      class="border-base-content border-b px-8 py-4"
    >
      <div
        class="text-base md:text-lg lg:text-xl"
      >
        Opponents
      </div>
    </div>
    <div
      class="w-full overflow-x-auto"
    >
      <table
        class="table"
      >
        <thead>
          <tr
            class="px-8"
          >
            <th
              align="left"
              class="pl-8"
            >
              No
            </th>
            <th>
              Opponent
            </th>
            <th
              align="right"
            >
              <button
                class="cursor-pointer capitalize"
                type="button"
              >
                Games
              </button>
            </th>
            <th
              align="right"
            >
              <div
                class="flex items-center justify-end gap-2"
              >
                <button
                  class="cursor-pointer capitalize"
                  type="button"
                >
                  Win
                </button>
                <button
                  class="cursor-pointer capitalize"
                  type="button"
                >
                  (%)
                </button>
              </div>
            </th>
            <th
              align="right"
            >
              <div
                class="flex items-center justify-end gap-2"
              >
                <button
                  class="cursor-pointer capitalize"
                  type="button"
                >
                  Draw
                </button>
                <button
                  class="cursor-pointer capitalize"
                  type="button"
                >
                  (%)
                </button>
              </div>
            </th>
            <th
              align="right"
              class="pr-8"
            >
              <div
                class="flex items-center justify-end gap-2"
              >
                <button
                  class="cursor-pointer capitalize"
                  type="button"
                >
                  Loss
                </button>
                <button
                  class="cursor-pointer capitalize"
                  type="button"
                >
                  (%)
                </button>
              </div>
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
